src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -403,8 +469,34 @@ def in_software_raid?(device)
         if device.is_a?(Planned::Device)
           device.is_a?(Planned::Md)
         else
-          device.ancestors.any? { |dev| dev.is?(:software_raid) }
+          device.ancestors.any? do |dev|
+            # Don't check boot_disk as it might validly be a RAID1 itself
+            # (full disks as RAID case) - we want to treat this as 'no RAID'.
+            dev.is?(:software_raid) && dev != boot_disk
+          end
+        end
+      end
+
+      # Check if device is a direct member of a RAID1 (RAID over entire disks).
+      #
+      # FIXME: The check is possibly overly strict: currently it enforces
+      #   that the disk is a member of a single RAID.
+      #   With BIOS RAIDs that might not be necessary. Also the limitation
```

Comment:
I don't get why BIOS RAIDs are mentioned here. We just treat them always as regular disks (they return true for `is?(:disk_device)`) and I don't see anything about BIOS RAIDs in this method, just software RAIDs are considered. Isn't it?

Reply:
I was thinking of:

https://github.com/openSUSE/libstorage-ng/blob/master/testsuite/probe/md-imsm1-devicegraph.xml

Reply:
That's my point. In that case, `/dev/md/a` and `/dev/md/b` are both BIOS RAIDs. So they are not affected by this method. Basically in all regards, both the proposal and the partitioner consider them as disks (because they return true for `is?(:disk_device)`). And they are skipped in the first line of this function loop (because they return false for `is?(:software_raid)`).

Unless I'm overlooking something.

Reply:
I find it less confusing now, without the reference to BIOS RAIDs.